.github/workflows/update-script.js

```diff
@@ -0,0 +1,158 @@
+  scriptPath: process.env.LOCAL_RUN_SCRIPT_PATH || './libs/deps/imslib.min.js',
+  origin: process.env.LOCAL_RUN_ORIGIN || 'origin',
+};
+
+const getPrDescription = ({ branch, scriptPath }) => `## Description
+Update ${scriptPath} to the latest version
+## Related Issue
+Resolves: NO TICKET - AUTOMATED CREATED PR.
+## Testing instructions
+1. Signing in should still function
+2. Signing out should still work
+3. Regression tests on all consumers
+## Test URLs
+**Acrobat:**
+- Before: https://www.stage.adobe.com/acrobat/online/sign-pdf.html?martech=off
+- After: https://www.stage.adobe.com/acrobat/online/sign-pdf.html?martech=off&milolibs=${branch}--milo--adobecom
+**BACOM:**
+- Before: https://business.stage.adobe.com/fr/customer-success-stories.html?martech=off
+- After: https://business.stage.adobe.com/fr/customer-success-stories.html?martech=off&milolibs=${branch}--milo--adobecom
+**CC:**
+- Before: https://main--cc--adobecom.hlx.live/?martech=off
+- After: https://main--cc--adobecom.hlx.live/?martech=off&milolibs=${branch}--milo--adobecom
+**Homepage:**
+- Before: https://main--homepage--adobecom.hlx.page/homepage/index-loggedout?martech=off
+- After: https://main--homepage--adobecom.hlx.page/homepage/index-loggedout?martech=off&milolibs=${branch}--milo--adobecom
+**Blog:**
+- Before: https://main--blog--adobecom.hlx.page/?martech=off
+- After: https://main--blog--adobecom.hlx.page/?martech=off&milolibs=${branch}--milo--adobecom
+**Milo:**
+- Before: https://main--milo--adobecom.hlx.page/ch_de/drafts/ramuntea/gnav-refactor?martech=off
+- After: https://${branch}--milo--adobecom.hlx.page/ch_de/drafts/ramuntea/gnav-refactor?martech=off`;
+
+const fetchScript = (path) => new Promise((resolve, reject) => {
+  console.log(`Fetching script from ${path}`);
+  https
+    .get(path, (res) => {
+      if (res.statusCode < 200 || res.statusCode >= 300) {
+        return reject(new Error(`statusCode=${res.statusCode}`));
+      }
+
+      let data = '';
+
+      res.on('data', (chunk) => {
+        data += chunk;
+      });
+
+      res.on('end', () => {
+        resolve({
+          data,
+          headers: res.headers,
+        });
+      });
+    })
+    .on('error', (err) => {
+      reject(err);
+    });
+});
+
+// Use this for conditional execution of commands
+const execSyncSafe = (command) => {
+  try {
+    execSync(command);
+  } catch (error) {
+    console.log(`Skipped command ${command}`);
+  }
+};
+
+const createAndPushBranch = ({ script, branch, scriptPath, origin = 'origin' }) => {
+  // When testing locally, u likely do not want to kill your dev branch
+  if (!localExecution) {
+    execSync('git config --global user.name "GitHub Action"');
+    execSync('git config --global user.email "action@github.com"');
+    execSync('git fetch');
+    execSync('git checkout stage');
+    execSyncSafe(`git branch -D ${branch}`);
+    execSync(`git checkout -b ${branch}`);
+  }
+  console.log('writing script to file', scriptPath);
```

Comment:
Should `imslib.min.js` be updated when running on local? I'm thinking the source of truth should be the upstream project to make sure we're consistent.